### ISSUE TITLE
undefined error shape

### ISSUE THREAD

**Issue by Tieske** [aws/aws-sdk-js#3449]
**Describe the bug**
The `marketplacecommerceanalytics-2015-07-01` definition references shape `MarketplaceCommerceAnalyticsException`, which is not in the `shapes` list

https://github.com/aws/aws-sdk-js/blob/master/apis/marketplacecommerceanalytics-2015-07-01.normal.json#L30

**Is the issue in the browser/Node.js?**
data definition

**SDK version number**
latest master

**To Reproduce (observed behavior)**
n.a.

**Expected behavior**
I would expect the "shapes" object to hold a definition for `MarketplaceCommerceAnalyticsException`

**Comment by ajredniwja** [comment]
Hey @Tieske thanks for reaching out to us, the APIs are generated by respective teams, is it possible for you to share your use case?

**Comment by Tieske** [comment]
@ajredniwja I'm do a lot of calls from OpenResty to the AWS API, hence figured to take the same approach and generate the functions from the metadata. But when parsing it, I found those elements missing.

I'm just at parsing for now, don't know whether I'll even use the error messages.

That said, seems bad practice to me to reference elements that do not exist. (after this one, I ran into at least another 20 or so)
